specification/dartLangSpec.tex

```diff
@@ -16245,18 +16246,20 @@ \subsection{Assignable Expressions}
 \LMLabel{assignableExpressions}
 
 \LMHash{}%
-Assignable expressions are expressions
+\IndexCustom{Assignable expressions}{assignable expression} are terms
 that can appear on the left hand side of an assignment.
-This section describes how to evaluate these expressions
+This section describes how to evaluate these terms
 when they do not constitute the complete left hand side of an assignment.
+The semantics of an assignment as a whole is described elsewhere
+(\ref{assignment}).
 
-\rationale{%
-Of course, if assignable expressions
-always appeared \emph{as} the left hand side,
-one would have no need for their value,
-and the rules for evaluating them would be unnecessary.
-However, assignable expressions can be subexpressions of other expressions
-and therefore must be evaluated.%
+\commentary{%
+An assignable expression can be a subterm of another term,
+in which case it may need to be evaluated to an object.
+For example, if we are assigning a value to an instance variable \code{x}
+of a receiver expression $e$ by means of the expression \code{$e$.x\,\,=\,\,42},
+we need to evaluate $e$
+such that we can update the \code{x} of the correct object.%
```

Comment:
Unclear that `e` is the assignable expression.
We are talking about assignable expressions, and the first line of the example introduces a variable `x`, which is the quintessential assignable expression, so the reader may easily jump to the conclusion that `x` is the assignable expression of this example. In fact it's `e`, which is never said to be an assignable expression.

(Would also like to expand on the "may" above, with an example of an assignable expression which is not evaluated.)

Maybe:
```latex
\commentary{%
An expression which is grammatically an assignable expression
can occur in a position where it's not assigned to, but is instead
used to refer to a value or scope, and may be evaluated to a value.

Because the grammar of assignable expressions includes the very general
single and qualified identifiers, an assignable expression can
have many different meanings, depending on binding of those identifiers
in the context.

For example, an expression \code{x.y} is grammatically an assignable expression.
That expression can occur as a subexpression of a larger expression, 
for example as code{x.y.z = 42}. 
In that position \code{x.y} may refer to a getter \code{y} on an object referenced by a variable \code{x},
in which case \code{x.y} will be evaluated to an object before accessing the `z` member of that object.
The code\{x.y} can also denote a class \code{y} referenced through an import prefix \code{x},
with \code{z} denoting a static variable of that class, 
in which case \code{x.y} will not be evaluated to a value.
}
```

Reply:
Done.